modules/account/lib/Account.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Jelix\Authentication\Account;
+
+class Account
+{
+    protected $name;
+
+    protected $email;
+
+    protected $status;
+
+    protected $create_date;
+
+    protected $provider;
+
+    protected $account_id;
+
+    public static $properties = array('name', 'email', 'status', 'create_date', 'provider', 'account_id');
+
+    public static $editableProperties = array('email');
+
+    /**
+     * @param \jDaoRecordBase $daoRecord The account dao entry
+     */
+    public function __construct($daoRecord)
+    {
+        foreach(self::$properties as $prop) {
+            if (property_exists($daoRecord, $prop)) {
+                $this->$prop = $daoRecord->$prop;
+            }
+        }
+    }
+
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    public function getEmail()
+    {
+        return $this->email;
+    }
+
+    public function getStatus()
+    {
+        return $this->status;
+    }
+
+    public function getCreate_date()
+    {
+        return $this->create_date;
+    }
+
+    public function getProvider()
+    {
+        return $this->provider;
+    }
+
+    public function getAccount_id()
+    {
+        return $this->account_id;
+    }
+
+    public function getInfosArray()
+    {
+        $infos = array();
+        $statuses = array('deleted', 'disabled', 'invalid', 'valid');
+        foreach (self::$properties as $prop) {
```

Comment:
instead of `self::$properties`, you could use `array_keys($daoRecord->getProperties())`.